packages/datadog-instrumentations/src/tedious.js

```diff
@@ -22,30 +21,26 @@ addHook({ name: 'tedious', versions: ['>=1.0.0'] }, tedious => {
       return makeRequest.apply(this, arguments)
     }
 
-    const callbackResource = new AsyncResource('bound-anonymous-fn')
-    const asyncResource = new AsyncResource('bound-anonymous-fn')
-
     const connectionConfig = this.config
+    const ctx = { queryOrProcedure, connectionConfig }
 
-    return asyncResource.runInAsyncScope(() => {
-      const payload = { queryOrProcedure, connectionConfig }
-      startCh.publish(payload)
-      queryParent[queryField] = payload.sql
+    return startCh.runStores(ctx, () => {
```

Comment:
@rochdev, pairing with Ruben on this PR, do we need `runStores` for both `startCh` and `finishCh`, the latter is handling the callback?

Reply:
Yes, when there is a callback involved then `runStores` is needed for `finish` so that the parent context can be restored for the callback.